### PR TITLE
Update Golang dependencies to Knative 1.0

### PR DESCRIPTION
**Pull request by skonto**
As per title. Also updates route patch to latest.

`./hack/update-deps.sh --upgrade` initially fails due to upstream release dep misalignment:
```
go get: knative.dev/operator@release-1.0 (v0.27.1) requires knative.dev/hack@v0.0.0-20211210083629-92d8a0a00cb6, not knative.dev/hack@release-1.0 (v0.0.0-20211122163517-fe1340f21191)
go get: knative.dev/operator@release-1.0 (v0.27.1) requires knative.dev/pkg@v0.0.0-20211210132429-e86584fd3c69, not knative.dev/pkg@release-1.0 (v0.0.0-20211101212339-96c0204a70dc)
```
had to set:
```
FLOATING_DEPS=(
  "knative.dev/networking@${KN_VERSION}"
  "knative.dev/operator@${KN_VERSION}"
)
```
after failure, update go.mod manually and then run again the deps
`./hack/update-deps.sh --upgrade`

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -22,10 +22,10 @@ SERVING_VERSION="release-v1.0.1"
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
 FLOATING_DEPS=(
-  "knative.dev/hack@${KN_VERSION}"
+
   "knative.dev/networking@${KN_VERSION}"
   "knative.dev/operator@${KN_VERSION}"
-  "knative.dev/pkg@${KN_VERSION}"
+
```
Comment (markusthoemmes) [inline hack/update-deps.sh]: Did you mean to commit this?

File: test/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -127,13 +127,11 @@ func preUpgradeTests() []pkgupgrade.Operation {
...
-		waitForServicesReady(ctx),
-		eventingupgrade.PostUpgradeTest(),
-		kafkaupgrade.ChannelPostUpgradeTest(),
+	tests := []pkgupgrade.Operation{waitForServicesReady(ctx)}
+	tests = append(tests, eventingupgrade.PostUpgradeTests()...)
+	tests = append(tests, kafkaupgrade.ChannelPostUpgradeTest(),
 		kafkaupgrade.SourcePostUpgradeTest(),
-		kafkabrokerupgrade.BrokerPostUpgradeTest(),
-	}
+		kafkabrokerupgrade.BrokerPostUpgradeTest())
```
Comment (markusthoemmes) [inline test/upgrade/upgrade_test.go:134]: reordering the tests is not an issue here, I'd assume. So I'd just delete the PostUpgradeTests from the initial list and append them like the serving tests.

File: hack/update-deps.sh (hunk 1 above)
Comment (skonto) [reply hack/update-deps.sh]: I restored in the above commit `restore deps.sh` it is not in?

File: test/upgrade/upgrade_test.go (hunk 2 above)
Comment (skonto) [reply test/upgrade/upgrade_test.go:134]: I was not sure about order.

Comment (skonto) [reply test/upgrade/upgrade_test.go:134]: @mgencur ordering makes a difference here?

File: hack/update-deps.sh (hunk 1 above)
Comment (markusthoemmes) [reply hack/update-deps.sh]: I might've screwed something up on my end in Github, sorry for the noise.

Comment (skonto) [reply hack/update-deps.sh]: np ;)

**Comment by skonto** [comment]
closing this in favor of #1354

**Pull request closed without merging**